src/extension.ts

```diff
@@ -102,6 +107,15 @@ export async function activate(context: vscode.ExtensionContext) {
   octokit = await credentials.getOctokit();
 
   vscode.window.onDidChangeTextEditorSelection(async (selection) => {
+    // // Get text of selected piece of code
+    let selectedCode = "";
+    if (selection.selections.length > 0) {
+      let selectedText = selection;
+      selectedCode= selectedText.textEditor.document.getText(selectedText.selections[0]);
```

Comment:
Interesting, my guess is that this would only select one line of code as only one param is passed but I might be wrong.
https://code.visualstudio.com/api/references/vscode-api#TextDocument

Reply:
It selects the whole block

Reply:
What do you mean by fold exactly?

Reply:
It was my understanding it got the full code block, a function, a class, a switch statement, or something like that, but I tested and it grabs the _selection_.

Reply:
Yeah, that's the idea. Sorry if I wasn't clear when I described it with words.